### PR TITLE
Stop caching sandbox images in analysis image.

### PR DESCRIPTION
**Pull request by oliverchang**
This causes various issues (such as compatibility on CI), and presents
scalability issues with larger images.

Instead, pull images on demand and cache them in persistent storage volume
on our workers. Make our workers a StatefulSet instead of Deployment to
support this.

This helps #146.

### CODE REVIEW

File: infra/worker/workers-set.yaml [hunk 1]
```diff
@@ -28,10 +29,23 @@ spec:
           value: firestore://projects/ossf-malware-analysis/databases/(default)/documents/
         securityContext:
           privileged: true
+        volumeMounts:
+          - mountPath: "/var/lib/containers"
+            name: image-storage
```
Comment (calebbrown) [inline infra/worker/workers-set.yaml:34]: Did you have any thoughts on how we can automate the invalidation of this cache overtime?

**Review by calebbrown — APPROVED** [review]

File: infra/worker/workers-set.yaml (hunk 1 above)
Comment (oliverchang) [reply infra/worker/workers-set.yaml:34]: We always pull the latest sandbox image here: https://github.com/ossf/package-analysis/blob/75b7264f7262505c6889e23b1a1f3e277812410b/internal/sandbox/sandbox.go#L92 so we should have the latest image.

But yeah, I think this does leave around stale layers over time. We may need to call `podman image prune` or somewhere.

Comment (oliverchang) [reply infra/worker/workers-set.yaml:34]: Added a call to that in `internal/sandbox/sandbox.go`

Comment (calebbrown) [reply infra/worker/workers-set.yaml:34]: nice!

**Pull request merged**